build_runner/lib/src/package_graph/package_graph.dart

```diff
@@ -147,8 +150,25 @@ class PackageGraph {
   /// last. For any two packages for which neither is a transitive dependency of
   /// the other the relative position of the packages within the cycle is
   /// non-deterministic.
+  @Deprecated('Use orderedPackageCycles instead')
   Iterable<PackageNode> get orderedPackages =>
-      _orderedPackages(root, new Set<PackageNode>());
+      orderedPackageCycles.expand((c) => c);
+
+  /// Collects packages into strongly connected subcomponents and orders those
+  /// components in reverse dependency order.
+  ///
+  /// A cycle is defined as a collection of packages where each package is
+  /// present in the transitive dependency graph of each other package.
+  ///
+  /// Cycles are ordered in reverse dependency order, so if package `A` depends
```

Comment:
`Cycles are ordered in reverse dependency order` I don't understand what you mean by this - maybe you really mean `components`? That seems duplicative with the first sentence though.